sandbox/Example.Core.SubscribeRaw/Program.cs

```diff
@@ -11,11 +11,11 @@
 
 Print($"[SUB] Subscribing to subject '{subject}'...\n");
 
-var sub = await connection.SubscribeAsync(subject);
+var sub = await connection.SubscribeAsync<byte[]>(subject);
 
 await foreach (var msg in sub.Msgs.ReadAllAsync())
 {
-    var data = Encoding.UTF8.GetString(msg.Data.ToArray());
+    var data = Encoding.UTF8.GetString(msg.Data!);
```

Comment:
Why is the [null-forgiving operator](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-forgiving) needed here?  Is `SubscribeAsync<T>` eventually returning `T?` instead?  Would be nice if we could avoid that

Reply:
Default values in general yield empty/sentinel payloads that's why there are `T?`s in a lot of places. Do you think we shouldn't allow nulls? How would we handle sending and receiving empty payloads?

Reply:
True, I guess there is no way to avoid it for reference types when `default` may be retruned.  Seeing as [System.Text.JSON even does it](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializer.deserialize?view=net-7.0#system-text-json-jsonserializer-deserialize-1(system-readonlyspan((system-byte))-system-text-json-serialization-metadata-jsontypeinfo((-0))))